Start testing with 2.077 upstream DMD

mihails-strasuns-sociomantic opened the pull request.

mihails-strasuns-sociomantic committed:
Fix variable declaration inside `switch`

Errors out in DMD 2.077

mihails-strasuns-sociomantic committed:
Different handling of va_start in new DMD

mathias-lang-sociomantic commented on src/ocean/text/json/Json.d:
```diff
@@ -890,10 +890,10 @@ class Json(T) : JsonParser!(T)
                 switch (val.type)
                 {
-                    T[64] tmp = void;
-
```
2.066 ? Also this is a (reported) bug. But yeah the workaround is legit.

mihails-strasuns-sociomantic replied:
Yep, I checked how things look with 2.066 as 2.077 is not yet available in d-apt

mathias-lang-sociomantic replied:
Let me ask again: 2.0**6**6 ? With two 6 ? :)

mihails-strasuns-sociomantic replied:
:crab:

mathias-lang-sociomantic commented on src/ocean/util/log/PeriodicTrace.d:
```diff
@@ -187,7 +187,10 @@ struct PeriodicTracer
+        static if (__VERSION__ > 2071)
+            va_start(ap, fmt);
+        else
+            va_start(ap, __va_argsave);
```
Well we'll switch to CT vararg in v4.0.0 so I'm not sure this fix is worth it, is it ?

mihails-strasuns-sociomantic replied:
Probably. It looks like https://github.com/sociomantic-tsunami/ocean/pull/93 will require some breaking changes to switch to vanilla DMD one way or another, so your point makes perfect sense.

mihails-strasuns-sociomantic replied:
FYI: I have kept it so we can enable travis testing immediately. Feel free to kill it when CT vararg takes over.

leandro-lucarella-sociomantic commented:
So? Merge or what?

mathias-lang-sociomantic commented:
First commit doesn't hurt, second seems useless IMHO since we seem to be heading towards upstream compatibility with V4.

mihails-strasuns-sociomantic commented:
Yep, needs rebase on v4. A bit of low prio right now because of infinite stream of major scheduler bugs :X I will update it after https://github.com/sociomantic-tsunami/ocean/pull/330 is merged

mihails-strasuns-sociomantic committed:
Adjust to deprecated netinet import

Upstream DMD/druntime has deprecated `core.sys.linux.sys.netinet.tcp` in
favor of `core.sys.linux.netinet.tcp` (moving all definitions there). D1
runtime only provides it via old module name and older versions of
dmd2 don't have it at all.

mihails-strasuns-sociomantic committed:
Access errno using setter syntax

D1 runtime exposes errno as a getter/setter function but it is not
guaranteed to be like that. C defines it as global variable and thus D
bindings may do the same, which was indeed done in later versions of
druntime.

mihails-strasuns-sociomantic committed:
Remove left-over convert dependency

mihails-strasuns-sociomantic committed:
D2 compatible null handling in old Layout

In D1 `null` on its own is of `void*` type. In D2 it has special
`typeof(null)` type, and also special `typeid(null)` type info.

mihails-strasuns-sociomantic committed:
Fix implicit string concatenation

mihails-strasuns-sociomantic committed:
Start testing with upstream 2.077 DMD

mihails-strasuns-sociomantic committed:
Always resolve getMsg(e) to e.message()

mihails-strasuns-sociomantic commented:
Updated to include all changes needed to build and test with upstream DMD as well as travis script adjustment to start doing so.

nemanja-boric-sociomantic commented:
Today is a good day.

nemanja-boric-sociomantic commented on src/ocean/transition.d:
```diff
@@ -553,45 +553,7 @@ unittest
-        });
-        return buffer;
-    }
+    return e.message();
```
:heart:

leandro-lucarella-sociomantic replied:
![RIP](https://vignette.wikia.nocookie.net/mortalengines/images/8/8e/Rip.jpg/revision/latest?cb=20140807221244)

mihails-strasuns-sociomantic commented:
Green!

mathias-lang-sociomantic commented on src/ocean/sys/HomeFolder.d:
```diff
@@ -187,7 +187,7 @@ version (Posix)
             // Obtain info from database.
             passwd *verify;
-            core.stdc.errno.errno(0);
+            core.stdc.errno.errno = 0;
```
Explanation would be nice

leandro-lucarella-sociomantic replied:
`sys/HomeFolder`. OMG, Tango had a whole module to get the user's home "folder" ("folder" is SOOO windows :roll_eyes: ).

Anyway...

mathias-lang-sociomantic commented on src/ocean/io/select/protocol/fiber/FiberSelectWriter.d:
```diff
@@ -43,6 +43,8 @@ import ocean.stdc.posix.netinet.in_: IPPROTO_TCP;
 static if (__VERSION__ >= 2000 && __VERSION__ < 2073)
     enum { TCP_CORK = 3 }
+else static if (__VERSION__ >= 2077)
+    import core.sys.linux.netinet.tcp: TCP_CORK;
```
An explanation on why `core.sys.linux.netinet.tcp` can't be used or the time (or `core.sys.linux.sys.netinet.tcp`) would be nice

leandro-lucarella-sociomantic replied:
Not sure if it's worth it, but maybe the commit comment can go in the code too.

mihails-strasuns-sociomantic commented:
Updated with more detailed commit text for @mathias-lang-sociomantic :)

mathias-lang-sociomantic reviewed (approved):
I'm happy

leandro-lucarella-sociomantic commented on src/ocean/text/convert/Layout_tango.d:
```diff
@@ -805,7 +805,13 @@ class Layout(T)
+        else
+            bool null_ti = tinfo.classinfo is TypeInfo.classinfo;
+
+        if (null_ti)
```
This might also need a better commit message at least (but if you think it is obvious is fine with me).

leandro-lucarella-sociomantic reviewed (approved):
Besides tiny comments, LGTM. Feel free to self-merge.

leandro-lucarella-sociomantic commented:
I don't get the coverage changes though.

mihails-strasuns-sociomantic commented:
@leandro-lucarella-sociomantic 

> I don't get the coverage changes though

Apart from "unexpected coverage" bit which seems completely broken now, everything else matches my expectations. Some of affected modules (like `HomeFolder`) are not tested, thus modifying them affects patch coverage negatively.

mihails-strasuns-sociomantic commented:
Updated with more explanations about `typeid(null)`
